appcues/src/test/java/com/appcues/data/remote/DataLogcuesTest.kt

```diff
@@ -0,0 +1,378 @@
+        // WHEN
+        dataLogcues.error("description", "test exception message")
+        // THEN
+        verifySequence {
+            logcues.error("description.\nReason: test exception message")
+        }
+    }
+
+    @Test
+    fun `error SHOULD log error pretty`() {
+        // GIVEN
+        val reason =
+            """
+            HttpError(code=401,    error=ErrorResponse(error="error (1), {2}, [3]", statusCode=401, message=ErrorMessageResponse()))
+            """.trimIndent()
+
+        // WHEN
+        dataLogcues.error("description", reason)
+        // THEN
+        verifySequence {
+            logcues.error(
+                """
+                description.
+                Reason: HttpError(
+                  code=401,
+                  error=ErrorResponse(
+                    error="error (1), {2}, [3]",
+                    statusCode=401,
+                    message=ErrorMessageResponse()
+                  )
+                )
+                """.trimIndent()
+            )
+        }
+    }
+
+    @Test
+    fun `debug request SHOULD log WHEN null body`() {
+        // GIVEN
+        val request = mockk<Request> {
+            every { method } returns "GET"
+            every { url } returns "http://www.appcues.com/".toHttpUrl()
+            every { headers } returns Headers.headersOf()
+            every { body } returns null
+        }
+        // WHEN
+        dataLogcues.debug(request)
+        // THEN
+        verifySequence {
+            logcues.debug(
+                """
+                REQUEST: GET
+                URL: http://www.appcues.com/
+                
+                (no request body)
+                """.trimln()
+            )
+        }
+    }
+
+    @Test
+    fun `debug request SHOULD log WHEN body isDuplex`() {
+        // GIVEN
+        val request = mockk<Request> {
+            every { method } returns "HEAD"
+            every { url } returns "http://www.appcues.com/".toHttpUrl()
+            every { headers } returns Headers.headersOf()
+            every { body } returns mockk {
+                every { isDuplex() } returns true
+            }
+        }
+        // WHEN
+        dataLogcues.debug(request)
+        // THEN
+        verifySequence {
+            logcues.debug(
+                """
+                REQUEST: HEAD
+                URL: http://www.appcues.com/
+                
+                (duplex request body omitted)
+                """.trimln()
+            )
+        }
+    }
+
+    @Test
+    fun `debug request SHOULD log WHEN body isOneShot`() {
+        // GIVEN
+        val request = mockk<Request> {
+            every { method } returns "HEAD"
+            every { url } returns "http://www.appcues.com/".toHttpUrl()
+            every { headers } returns Headers.headersOf()
+            every { body } returns mockk {
+                every { isDuplex() } returns false
+                every { isOneShot() } returns true
```

Comment:
nice catch on handling these special cases in logging, had not thought about that

Reply:
this is all inspired on the original HttpLoggingInterceptor, a lesser scope since we dont need to support every configurable option they have.